api/v1alpha1/openstackrouter_types.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OpenStackRouterSpec defines the desired state of OpenStackRouter
+type OpenStackRouterSpec struct {
+	// Cloud is the OpenStackCloud hosting this resource
+	Cloud string `json:"cloud"`
+
+	// ID is the OpenStack UUID of the resource. If left empty, the
+	// controller will create a new resource and populate this field. If
+	// manually populated, the controller will adopt the corresponding
+	// resource.
+	ID string `json:"id,omitempty"`
+
+	// Name of the OpenStack resource.
+	Name string `json:"name,omitempty"`
+
+	Description string `json:"description,omitempty"`
+
+	ExternalGatewayInfo OpenStackRouterExternalGatewayInfo `json:"externalGatewayInfo,omitempty"`
```

Comment:
This is optional. The type should be a pointer, so that it's easier to know whether the user actually passed one

```suggestion
	ExternalGatewayInfo *OpenStackRouterExternalGatewayInfo `json:"externalGatewayInfo,omitempty"`
```

Reply:
:point_up:

Reply:
Yes, I am working on a PR to solve some issues at the same time.